presto-parser/src/main/java/com/facebook/presto/sql/tree/BinaryLiteral.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.sql.tree;
+
+import com.facebook.presto.sql.parser.ParsingException;
+import com.google.common.io.BaseEncoding;
+import io.airlift.slice.Slice;
+import io.airlift.slice.Slices;
+
+import java.util.Optional;
+
+import static java.util.Objects.requireNonNull;
+
+public class BinaryLiteral
+        extends Literal
+{
+    private final String value;
+    private final Slice slice;
+
+    public BinaryLiteral(String value)
+    {
+        this(Optional.empty(), value);
+    }
+
+    public BinaryLiteral(Optional<NodeLocation> location, String value)
+    {
+        super(location);
+        requireNonNull(value, "value is null");
+        // the grammar could possibly include White Space in the value it passes to us
+        this.value = value.replaceAll("[ \\r\\n\\t]", "").toUpperCase();
```

Comment:
Since we're not preserving the original string as written by the user, what's the purpose of keeping the `String` variant of value? In that case, keeping the binary representation (i.e., `Slice`) would be sufficient.